src/GitHub.Api/Git/Repository.cs

```diff
@@ -120,21 +191,139 @@ public ITask Revert(string changeset)
             return repositoryManager.Revert(changeset);
         }
 
-        public ITask ListLocks()
+        public ITask RequestLock(string file)
         {
-            if (repositoryManager == null)
-                return new ActionTask(new NotReadyException().ToTask<bool>());
-            return repositoryManager.ListLocks(false);
+            return repositoryManager.LockFile(file).Then(UpdateLocks);
```

Comment:
For readability's sake, `Then` actions should be on their own line, they're pretty easy to miss otherwise

Reply:
👍